design-documents/storefront/catalog/product-options-and-variants.md

```diff
@@ -0,0 +1,355 @@
+
+## Document purpose
+
+Product options are a powerful instrument that allows a shopper to customize and (or) personalize a product before adding the product to the cart. 
+
+Introduction of the storefront API brings us a unique opportunity to revisit the options management, address the existing limitation, and address known issues:
+
+* Reduce redundancy caused by variants matrix creation. Magento creates variant and simple product for each intersection of options and option values selected to represent the configurable product. Not all the cases in real require a product creation, so extracting the variant as an entity could reduce system load caused by the number of products. 
```

Comment:
Please add a section that will describe a data migration plan/algorithm for Magento merchants. 
- How to reduce the number of variations taking into account that Magento already has separate products for variations?
- Where everything is going to happen(Message broker or Monolith)?
- The shape on monolith APIs

Reply:
the first iteration does not assume the entities reduction, since at the first round we will have to support Magento cases only, but idea behind that statement is that for future you may have API and implementation that may share the same variant across the different option selection. At this stage, we are defining APIs which do not contradict with the mentioned behavior.